Update cloud cli docs with windows, new instructions (#4172)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/cloud-cli-installation.md

dbt commands are run against dbt Cloud's infrastructure and benefit from:

* Secure credential storage in the dbt Cloud platform.
* Automatic deferral of build artifacts to your Cloud project's production environment. 
* Speedier, lower cost builds.
* Significant platform improvements, to be released over the coming months.

After installing, you invoke the Cloud CLI the same way you would call dbt Core.  For example, to compile a project using dbt Cloud:

```bash
dbt deps
dbt compile
```

## Installing the dbt Cloud CLI

:::caution

For compatibility, both the Cloud CLI and dbt Core are invoked by running `dbt`. This can cause a path conflict, where your OS invokes either the Cloud CLI or dbt Core depending on it's presence in your $PATH environment variable.

If you have dbt Core installed locally, ensure that you either deactivate your Python environment or uninstall before proceeding.  Alternatively, advanced users can modify the $PATH environment variable to correctly point to the dbt Cloud CLI binary to use both the Cloud CLI and dbt Core together.

You can always uninstall the Cloud CLI to return to using dbt Core.
:::

### Install and update with Brew - MacOS

> 

Reviewer: given homebrew isn’t going to be supported for core, is this still right? do we need to say anything about [bundles](https://github.com/dbt-labs/dbt-core-bundles)?

Author: This is still right for Cloud CLI. We might change to recommending manual installs for all platforms, but for now it's right.

1. Verify that there is no conflict with a dbt Core installation on your system. 

```bash
which dbt
# Should return "dbt not found"
```

If the dbt help text appears, deactivate your current dbt Core install.

2. Install the dbt Cloud CLI with Brew: 

> 

Reviewer: do users need to install homebrew first? http://brew.sh/

Author: Yes, but I haven't seen any struggles with that step. I think we could add a link to brew?

```bash
brew tap dbt-labs/dbt-cli
brew install dbt-cloud-cli
```

3. Verify the installation. If the help text does not indicate that you're using the dbt Cloud CLI, make sure you've deactivated your pyenv or venv and do not have a version of dbt globally installed.

```bash
dbt --help
```

#### Updating your dbt Cloud installation - MacOS

Run `brew upgrade dbt-cloud-cli` to update the Cloud CLI.  During the beta period, we recommend updating before filing a bug report, as the API is subject to breaking changes!

### Manually install - Windows

1. Download the latest Windows release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases).

> 2. Extract the `dbt-cloud-cli.exe` file into the same folder as your dbt project.

Reviewer: since there are many options, how do users know what release asset to download? for instance, if they have windows, do they download dbt_0.29.9_windows_amd64.tar.gz, or dbt_0.29.9_windows_arm64.tar.gz, or dbt_0.29.9_windows_armv6.tar.gz? when i downloaded those three, i didn't see 'dbt-cloud-cli.exe'. i saw 'dbt.exe'.

Author: ah, yes, this should be dbt.exe!

you pick the file based on your CPU architecture. The vast majority will want the amd64 version, as theyre likely running AMD/Intel CPUs.

> 2. Extract the `dbt-cloud-cli.exe` file into the same folder as your dbt project.

Reviewer: do they need to run any commands to verify it's installed? 

Run the following command to verify the installation: dbt --help

If the help text doesn't show you're using the dbt Cloud CLI, ensure you've deactivated pyenv or venv and don't have a global dbt version installed

Author: yeah, we should add that here too.

:::info

Advanced users can configure multiple projects to use the same Cloud CLI executeable by placing the executeable in the Program Files folder and [adding the Cloud CLI executeable to their Windows PATH environment variable](https://medium.com/@kevinmarkvi/how-to-add-executables-to-your-path-in-windows-5ffa4ce61a53).

Note that if you are using VS Code, you'll need to restart your IDE to pick up modified environment variables.

> :::

Reviewer: should we not use ide since we have a dbt cloud ide? should we instead say:

If you're using VS Code, remember to restart it so it can pick up modified environment variables.

#### Updating your dbt Cloud installation - Windows

Follow the same process in [Installing dbt Cloud CLI](#manually-install-windows-only) and replace the existing `dbt` executable with the new one. During the beta period, we recommend updating before filing a bug report, as the API is subject to breaking changes!

### Manually install - Linux 

> 1. Download the latest Linux release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases).

Reviewer: same question as above

Author: Still CPU architecture based - Linux users should be relatively used to doing this.

```bash
dbt --help
```

## Setting up the CLI - MacOS and Linux

Once installed, we need to configure the Cloud CLI to connect to a dbt Cloud project.

1. Ensure that you have created a project in [dbt Cloud](https://cloud.getdbt.com/).

2. Ensure that your personal [development credentials](https://cloud.getdbt.com/settings/profile/credentials) are set on the project. The Cloud CLI will use these credentials, stored securely in dbt Cloud, to communicate with your data warehouse.

3. Navigate to [your profile](https://cloud.getdbt.com/settings/profile) and enable the **Beta** flag under **Experimental Features.**

4. Create an environment variable with your [dbt Cloud API key](https://cloud.getdbt.com/settings/profile#api-access):

On MacOS, Linux, or Windows add an environment variable:
```bash
export DBT_CLOUD_API_KEY="1234" # Replace 1234 with your API key   
```

In Powershell, add an environment variable:

Note that this variable will be reset if you restart your shell. To add an environment variable permanently, add a system environment variable in your platform.

5. Navigate to a dbt project

```bash
cd ~/dbt-projects/jaffle_shop
```

> 6. In your `dbt_project.yml` file, ensure there is a section titled "Cloud". This section is required to have a `project-id` field with a valid [project ID](#glossary). 

Author: This is not implemented yet, so we're blocked from merging.

Author: This is now implemented - we should merge ASAP!

```yaml
# dbt_project.yml
name:

version:
...

cloud:
    project-id: PROJECT_ID
```

You can find your project ID by navigating to dbt Cloud, selecting your project and clicking on **Develop** in the navigation bar. Your project ID is the number in the URL: https://cloud.getdbt.com/develop/26228/projects/PROJECT_ID.

## Using dbt Cloud CLI

The dbt Cloud CLI is a drop-in replacement for dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. Since this is still invoking dbt under the hood, you'll still need to run `dbt deps`, followed by the same model build commands you typically run.

Have features you'd like to see in the Cloud CLI? Reach out to us on Slack!

## Glossary

> 

Reviewer: is it replace `dbt` or replace `dbt-cloud-cli.exe`?

---
title: Install dbt Cloud CLI 
id: cloud-cli-installation
description: "Instructions for installing and configuring dbt Cloud CLI"
pagination_next: "docs/cloud/configure-cloud-cli"
---

:::info Beta functionality

The dbt Cloud CLI is currently in [public beta](/docs/dbt-versions/product-lifecycles#dbt-cloud). Share feedback or request features you'd like to see on the [dbt community Slack](https://getdbt.slack.com/archives/C05M77P54FL).

::: 

dbt Cloud natively supports developing using a command line (CLI), empowering team members to contribute with enhanced flexibility and collaboration. The dbt Cloud CLI allows you to run dbt commands against your dbt Cloud development environment from your local command line.

dbt commands are run against dbt Cloud's infrastructure and benefit from:

* Secure credential storage in the dbt Cloud platform.
* Automatic deferral of build artifacts to your Cloud project's production environment. 
* Speedier, lower cost builds.

> * Significant platform improvements, to be released over the coming months.

Reviewer: ```suggestion
* Support for dbt Mesh (cross-project `ref`),
* Significant platform improvements, to be released over the coming months.
```

> The dbt Cloud CLI, powered by [dbt Cloud](/docs/cloud/about-cloud/dbt-cloud-features), is distinct from [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source command-line tool for dbt. Both tools are command lines that use the same dbt commands, but the dbt Cloud CLI is tailored to run specifically on dbt Cloud's infrastructure.

Reviewer: ```suggestion
The dbt Cloud CLI and [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source project, are both command line tools that let you run your dbt projects and use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure.
```

---
title: Install dbt Cloud CLI 
id: cloud-cli-installation
description: "Instructions for installing and configuring dbt Cloud CLI"
pagination_next: "docs/cloud/configure-cloud-cli"
---

:::info Beta functionality

The dbt Cloud CLI is currently in [public beta](/docs/dbt-versions/product-lifecycles#dbt-cloud). Share feedback or request features you'd like to see on the [dbt community Slack](https://getdbt.slack.com/archives/C05M77P54FL).

::: 

dbt Cloud natively supports developing using a command line (CLI), empowering team members to contribute with enhanced flexibility and collaboration. The dbt Cloud CLI allows you to run dbt commands against your dbt Cloud development environment from your local command line.

dbt commands are run against dbt Cloud's infrastructure and benefit from:

* Secure credential storage in the dbt Cloud platform.
* Automatic deferral of build artifacts to your Cloud project's production environment. 
* Speedier, lower cost builds.
* Significant platform improvements, to be released over the coming months.

> Note, the dbt Cloud CLI is a separate tool powered by [dbt Cloud](/docs/cloud/about-cloud/dbt-cloud-features). It differs from [dbt Core](https://github.com/dbt-labs/dbt-core), which provides dbt through its open-source command-line tool.

Reviewer: ADD INFO HOW MUST HAVE DBT CLOUD ACCOUNT TO USE (MAYBE NEED PREREQS)

> 

Reviewer: greg to verify commands are the same (maybe a few not needed in cloud cli, like dbt init)

Author: Nearly all are the exact same. The only difference I know of, as of now, is that you need to quote selection syntax - https://dbtlabs.atlassian.net/browse/CLI-140

Author: Nearly all are the exact same. The only difference I know of, as of now, is that you need to quote selection syntax - https://dbtlabs.atlassian.net/browse/CLI-140

Reviewer: gotcha! i've addressed this in the main pr: https://github.com/dbt-labs/docs.getdbt.com/pull/4173

#### Updating your dbt Cloud installation - Windows

Follow the same process in [Installing dbt Cloud CLI](#manually-install---windows) and replace the existing `dbt.exe` executable with the new one. During the beta period, we recommend updating before filing a bug report, as the API is subject to breaking changes!

### Manually install - Linux 

1. Download the latest Linux release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases). (Pick the file based on your CPU architecture)

2. Extract the `dbt-cloud-cli` binary to the same folder as your dbt project.

```bash
tar -xf dbt_0.29.9_linux_amd64.tar.gz
./dbt --version
```

:::info

Advanced users can configure multiple projects to use the same Cloud CLI executeable by adding it to their PATH environment variable in their shell profile.

:::

3. Run the following command to verify the installation:

```bash
./dbt --help
```

- If the help text doesn't show you're using the dbt Cloud CLI, ensure you've deactivated pyenv or venv and don't have a global dbt version installed

> ## Update dbt Cloud CLI

Reviewer: any manual update for linux?

Author: Same as windows - redownload via the link

<Tabs>

<TabItem value="brew" label="macOS">

Before you begin, make sure you have [Homebrew installed](http://brew.sh/) in your code editor or command line terminal. If your operating system runs into path conflicts, refer to the [FAQs](#faqs).

1. Run the following command to verify that there is no conflict with a dbt Core installation on your system:
   
```bash
which dbt
```
   - This should return a `dbt not found`. If the dbt help text appears, use `pip uninstall dbt` to deactivate dbt Core from your machine.
  
2. Install the dbt Cloud CLI with Homebrew: 

```bash
brew tap dbt-labs/dbt-cli
brew install dbt-cloud-cli
```

3. Verify the installation by running `dbt --help` from the command line. If the help text doesn't indicate that you're using the dbt Cloud CLI, make sure you've deactivated your pyenv or venv and don't have a version of dbt globally installed.

</TabItem>

<TabItem value="windows" label="Windows">

If your operating system runs into path conflicts, refer to the [FAQs](#faqs).

1. Download the latest Windows release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases).

> 2. Extract the `dbt.exe` executeable into the same folder as your dbt project.

Reviewer: ```suggestion
2. Extract the `dbt.exe` executable into the same folder as your dbt project.
```

:::info

Advanced users can configure multiple projects to use the same dbt Cloud CLI by placing the executeable in the Program Files folder and [adding it to their Windows PATH environment variable](https://medium.com/@kevinmarkvi/how-to-add-executables-to-your-path-in-windows-5ffa4ce61a53).

Note that if you are using VS Code, you'll need to restart it to pick up modified environment variables.
:::

3. Verify the installation by running `./dbt --help` from the command line. If the help text doesn't indicate that you're using the dbt Cloud CLI, make sure you've deactivated your pyenv or venv and don't have a version of dbt globally installed.

</TabItem>

<TabItem value="linux" label="Linux">

If your operating system runs into path conflicts, refer to the [FAQs](#faqs).

1. Download the latest Linux release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases). (Pick the file based on your CPU architecture)

2. Extract the `dbt-cloud-cli` binary to the same folder as your dbt project.

```bash
tar -xf dbt_0.29.9_linux_amd64.tar.gz
./dbt --version
```

:::info

> Advanced users can configure multiple projects to use the same Cloud CLI executeable by adding it to their PATH environment variable in their shell profile.

Reviewer: ```suggestion
Advanced users can configure multiple projects to use the same Cloud CLI executable by adding it to their PATH environment variable in their shell profile.
```

---
title: Install dbt Cloud CLI 
id: cloud-cli-installation
description: "Instructions for installing and configuring dbt Cloud CLI"
pagination_next: "docs/cloud/configure-cloud-cli"
---

:::info Public preview functionality

The dbt Cloud CLI is currently in [public preview](/docs/dbt-versions/product-lifecycles#dbt-cloud). Share feedback or request features you'd like to see on the [dbt community Slack](https://getdbt.slack.com/archives/C05M77P54FL).

::: 

dbt Cloud natively supports developing using a command line (CLI), empowering team members to contribute with enhanced flexibility and collaboration. The dbt Cloud CLI allows you to run dbt commands against your dbt Cloud development environment from your local command line.

dbt commands are run against dbt Cloud's infrastructure and benefit from:

* Secure credential storage in the dbt Cloud platform.
* Automatic deferral of build artifacts to your Cloud project's production environment. 
* Speedier, lower cost builds.
* Support for dbt Mesh (cross-project `ref`),
* Significant platform improvements, to be released over the coming months.

> The dbt Cloud CLI and [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source project, are both command line tools that let you run your dbt projects and use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure.

Reviewer: ```suggestion
The dbt Cloud CLI and [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source project, are both command line tools that let you run your dbt projects and use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure and integrates those builds with all [dbt Cloud features](/docs/cloud/about-cloud/dbt-cloud-features).
```

:::

3. Verify the installation by running `./dbt --help` from the command line. If the help text doesn't indicate that you're using the dbt Cloud CLI, make sure you've deactivated your pyenv or venv and don't have a version of dbt globally installed.
-->
## Update dbt Cloud CLI

The following instructions explain how to update the dbt CLoud CLI to the latest version depending on your operating system. During the public preview period, we recommend updating before filing a bug report. This is because the API is subject to breaking changes.

#### Update with Homebrew (MacOS)

To update the dbt Cloud CLI, run `brew upgrade dbt-cloud-cli`. 
#### Update manually (Windows and Linux)

To update, follow the same process explained in [Install manually (Windows)](#install-manually-windows) and replace the existing `dbt.exe` executable with the new one.

## Next steps

After installation, you can [configure](/docs/cloud/configure-cloud-cli) the dbt Cloud CLI for your dbt Cloud project and use it to run [dbt commands](/reference/dbt-commands) similar to dbt Core. For example, you can execute the following commands to compile a project using dbt Cloud:

```bash

> dbt deps

Reviewer: removing per @greg-mckeon 
```suggestion
```

---
title: Install dbt Cloud CLI 
id: cloud-cli-installation
description: "Instructions for installing and configuring dbt Cloud CLI"
pagination_next: "docs/cloud/configure-cloud-cli"
---

> :::info Public preview functionality

Reviewer 2: Should this be a reusable snippet so we can keep it in sync with the other one?

You can install the dbt Cloud CLI on the command line by using one of these methods:

<Tabs queryString="install">

<TabItem value="pip" label="Existing dbt Core users (pip)">

:::info Use native packages or a virtual environment to avoid overriding dbt Core

One of the benefits of the dbt Cloud CLI is that there is no need to manage Python environments. We provide this Python package for users who are using the dbt Cloud CLI in place of dbt Core for easy compatibility with existing environments.

> It replaces dbt Core, but this change can be avoided by using the native install method and configuring your PATH or by creating a new virtual environment. To switch back to dbt Core, follow the dbt Core installation instructions.

Author: Could we delete the first two sentences, and just say "Installing the dbt Cloud CLI with pip replaces dbt Core. This change can be avoided by using the native install method and configuring your PATH or by creating a new virtual environment. 

Otherwise, to switch back to dbt Core, uninstall the Cloud CLI and follow the dbt Core installation instructions."

::: 

Before installing the dbt Cloud CLI, make sure you have Python installed and your virtual environment venv or pyenv . If you already have a Python environment configured, you can skip to the [pip installation step](#install-dbt-cloud-cli-in-pip).

### Install a virtual environment

> We recommend using virtual environments (venv) to namespace pip modules.

Author: "to namespace the cloud cli"

1. Create a new venv:
   ```shell

>    python3 -m venv dbt-env

Author: I would name this venv "dbt-cloud" so that there isn't confusion if they have followed the core install docs.

So: 

python3 -m venv dbt-cloud

    ```

2. Activate the virtual environment each time you create a shell window or session:
  ```shell   

>   source dbt-env/bin/activate         # activate the environment for Mac and Linux OR

Author: source dbt-cloud/bin/activate

  dbt-env\Scripts\activate            # activate the environment for Windows
  ```

> 3. (Mac and Linux only) Create an alias to activate your dbt environment with every new shell window or session. You can add the following to your shell's configuration file (for example, $HOME/.bashrc, $HOME/.zshrc) while replacing `<PATH_TO_VIRTUAL_ENV_CONFIG>` with the path to your virtual environment configuration:

Reviewer: per benoit, this is for mac and linux - what is step for windows?

Author: We don't include an alias in our other docs for venv, because it's difficult to explain and many users won't have permissions to do so. I think we can leave it off here.

One of the benefits of the dbt Cloud CLI is that there is no need to manage Python environments. We provide this Python package for users who are using the dbt Cloud CLI in place of dbt Core for easy compatibility with existing environments.

It replaces dbt Core, but this change can be avoided by using the native install method and configuring your PATH or by creating a new virtual environment. To switch back to dbt Core, follow the dbt Core installation instructions.

::: 

Before installing the dbt Cloud CLI, make sure you have Python installed and your virtual environment venv or pyenv . If you already have a Python environment configured, you can skip to the [pip installation step](#install-dbt-cloud-cli-in-pip).

### Install a virtual environment

We recommend using virtual environments (venv) to namespace pip modules.

1. Create a new venv:
   ```shell
   python3 -m venv dbt-env
    ```

2. Activate the virtual environment each time you create a shell window or session:
  ```shell   
  source dbt-env/bin/activate         # activate the environment for Mac and Linux OR
  dbt-env\Scripts\activate            # activate the environment for Windows
  ```

> 3. Create an alias to activate your dbt environment with every new shell window or session. You can add the following to your shell's configuration file (for example, $HOME/.bashrc, $HOME/.zshrc) while replacing `<PATH_TO_VIRTUAL_ENV_CONFIG>` with the path to your virtual environment configuration:

Reviewer: step 3 if for mac/linux only -- what about windows?

  ```

### Install dbt Cloud CLI in pip

1. (Optional) If you already have dbt Core installed, this installation will override that package. Note your dbt Core version in case you need to reinstall it later:

  ```bash
  dbt --version
  ```

2. Run the following command to install the dbt Cloud CLI:

  ```bash
  pip3 install dbt
  ```

4. (Optional) To revert back to dbt Core, first uninstall both the dbt Cloud CLI and dbt Core
5. Reinstall dbt Core using the version from Step 2.

  ```bash
  pip3 uninstall dbt-core dbt
  pip3 install dbt-core==VERSION
  ```

</TabItem>
	
<TabItem value="brew" label="macOS (brew)">

Before you begin, make sure you have [Homebrew installed](http://brew.sh/) in your code editor or command line terminal. Refer to the [FAQs](#faqs) if your operating system runs into path conflicts.

1. Run the following command to verify that there is no conflict with a dbt Core installation on your system:
   
  ```bash
  which dbt
  ```
   - This should return a `dbt not found`. If the dbt help text appears, use `pip uninstall dbt` to deactivate dbt Core from your machine.
  
2. Install the dbt Cloud CLI with Homebrew: 

  ```bash
  brew tap dbt-labs/dbt-cli

>   brew install dbt

Reviewer: per benoit, this command should be same command as update one:

`brew install dbt` or `brew upgrade dbt-cloud-cli` -- is it dbt-cloud-cli or dbt?

Author: it is dbt. this one is correct

:::

3. Verify the installation by running `./dbt --help` from the command line. If the help text doesn't indicate that you're using the dbt Cloud CLI, make sure you've deactivated your pyenv or venv and don't have a version of dbt globally installed.
  * You no longer need to use the `dbt deps` command. Previously, you had to run that command.

</TabItem>

> <TabItem value="linux" label="Linux (native executeable)">

Reviewer: ```suggestion
<TabItem value="linux" label="Linux (native executable)">
```

:::

3. Verify the installation by running `./dbt --help` from the command line. If the help text doesn't indicate that you're using the dbt Cloud CLI, make sure you've deactivated your pyenv or venv and don't have a version of dbt globally installed.
  * You no longer need to use the `dbt deps` command. Previously, you had to run that command.

</TabItem>

</Tabs>

## Update dbt Cloud CLI

The following instructions explain how to update the dbt CLoud CLI to the latest version depending on your operating system. During the public preview period, we recommend updating before filing a bug report. This is because the API is subject to breaking changes.

<Tabs>

<TabItem value="existing" label="Existing dbt Core users (pip)">

To update, run `pip install --upgrade dbt`.
	
<TabItem value="mac" label="macOS (brew)">

To update the dbt Cloud CLI, run `brew upgrade dbt-cloud-cli`. 

</TabItem>

> <TabItem value="windowslinux" label="Windows and Linux (executeables)">

Reviewer: ```suggestion
<TabItem value="windowslinux" label="Windows and Linux (executable)">
```

:::

3. Verify the installation by running `./dbt --help` from the command line. If the help text doesn't indicate that you're using the dbt Cloud CLI, make sure you've deactivated your pyenv or venv and don't have a version of dbt globally installed.
   * You don't have to run the `dbt deps` command when your environment starts. Previously, you had to do it during initialization. However, you'll still need to run `dbt deps` if you make changes to your `packages.yml` file.

</TabItem>

</Tabs>

## Update dbt Cloud CLI

The following instructions explain how to update the dbt CLoud CLI to the latest version depending on your operating system. 

During the public preview period, we recommend updating before filing a bug report. This is because the API is subject to breaking changes.

<Tabs>
<TabItem value="existing" label="Existing dbt Core users (pip)">

To update:
- Make sure you're in your virtual environment
- Run `pip install --upgrade dbt`.
	
</TabItem>

<TabItem value="mac" label="macOS (brew)">

> To update the dbt Cloud CLI, run `brew upgrade dbt-cloud-cli`. 

Reviewer: per benoit:

- this command should be same command as intall one: `brew install dbt` vs `brew upgrade dbt-cloud-cli ` -- is it `dbt` or `dbt-cloud-cli`?
- to update in brew, users can run `brew install` to install latest version. is it `brew upgrade` or `brew install`?

Author: either works, but I believe the standard is brew upgrade. it should also be `brew install dbt`, good catch

   * You don't have to run the `dbt deps` command when your environment starts. Previously, you had to do it during initialization. However, you'll still need to run `dbt deps` if you make changes to your `packages.yml` file.

</TabItem>

</Tabs>

## Update dbt Cloud CLI

The following instructions explain how to update the dbt CLoud CLI to the latest version depending on your operating system. 

During the public preview period, we recommend updating before filing a bug report. This is because the API is subject to breaking changes.

<Tabs>
<TabItem value="existing" label="Existing dbt Core users (pip)">

To update:
- Make sure you're in your virtual environment
- Run `pip install --upgrade dbt`.
	
</TabItem>

<TabItem value="mac" label="macOS (brew)">

To update the dbt Cloud CLI, run `brew upgrade dbt-cloud-cli`. 

</TabItem>

<TabItem value="windows" label="Windows (executable)">

To update, follow the same process explained in [Windows](/docs/cloud/cloud-cli-installation?install=windows#install-dbt-cloud-cli) and replace the existing `dbt.exe` executable with the new one.

</TabItem>

<TabItem value="linux" label="Linux (executable)">

> To update, follow the same process explained in [Windows](/docs/cloud/cloud-cli-installation?install=linux#install-dbt-cloud-cli) and replace the existing `dbt` executable with the new one.

Reviewer: per Benoit, linux just uses `dbt` file

Author: yes, that is right. the file is named `dbt` instead of `dbt.exe`
